src/main/java/seedu/address/model/tut/Tutorial.java

```diff
@@ -11,182 +11,402 @@
+     */
+    public static Tutorial of(TutName tutName, TutorialClass tutorialClass) {
+        requireNonNull(tutName);
+        requireNonNull(tutorialClass);
+        return Tutorial.exist(tutName, tutorialClass);
     }
 
     /**
-     * Marks the attendance for the student for the particular tutorial date.
+     * Represents a None type of tutorial (non-existent).
      */
-    public void markAttendance(Student student, TutDate tutorialDate) throws TutDateNotFoundException {
-        if (!isValidTutorialDate(tutorialDate)) {
-            throw new TutDateNotFoundException();
+    private static final class None extends Tutorial {
+        private static final None none = new None();
+
+        private final List<Student> students = new ArrayList<>();
+
+        public static None none() {
+            return none;
         }
-        if (!studentInList(student)) {
-            add(student);
+
+        @Override
+        public void add(Student student) {
+            if (student != null && !students.contains(student)) {
+                students.add(student);
+            }
+        }
+
+        @Override
+        public List<Student> getStudents() {
+            return this.students;
+        }
+
+        @Override
+        public boolean setAttendance(Date date, StudentId target) {
+            throw new NoTutorialException();
+        }
+
+        @Override
+        public void assignStudent(Student student) {
+            requireNonNull(student);
+            students.add(student);
+        }
+
+        @Override
+        public TutName getTutName() {
+            throw new NoTutorialException();
+        }
+
+        @Override
+        public Student get(Name name) {
+            throw new NoTutorialException();
+        }
+
+        @Override
+        public boolean tutorialDateInList(TutDate date) {
+            throw new NoTutorialException();
+        }
+
+        @Override
+        public void addTutorialDate(TutDate tutorialDate) {
+            throw new NoTutorialException();
+        }
+
+        @Override
+        public TutDate getTutorialDate(Date date) {
+            throw new NoTutorialException();
+        }
+
+        @Override
+        public boolean isValidTutorialDate(TutDate tutorialDate) {
+            throw new NoTutorialException();
+        }
+
+        @Override
+        public List<TutDate> getTutDates() {
+            throw new NoTutorialException();
+        }
+
+        @Override
+        public TutorialClass getTutorialClass() {
+            throw new NoTutorialException();
+        }
+
+        @Override
+        public void markAttendance(Student student, TutDate tutorialDate) throws TutDateNotFoundException {
+            throw new NoTutorialException();
+        }
+
+        @Override
+        public boolean studentInList(Student student) {
+            return students.contains(student);
+        }
+
+        @Override
+        public void deleteStudent(Student student) {
```

Comment:
Should this method throw studentNotFound exception if the student is not in the list? Just to remind the user or programmer in case there is some typo or issues.